language/stdlib/modules/DesignatedDealer.move

```diff
@@ -114,7 +114,7 @@ module DesignatedDealer {
         assert(Roles::has_treasury_compliance_role(tc_account), EACCOUNT_NOT_TREASURY_COMPLIANCE);
         let tiers = &mut borrow_global_mut<TierInfo<CoinType>>(dd_addr).tiers;
         let number_of_tiers = Vector::length(tiers);
-        assert(number_of_tiers + 1 <= MAX_NUM_TIERS, EINVALID_TIER_ADDITION);
+        assert(number_of_tiers <= MAX_NUM_TIERS, EINVALID_TIER_ADDITION);
```

Comment:
This change does not look correct to me.  tiers has not been updated yet.  We have to make sure once a tier is added, the number of tiers continues to be bounded by MAX_NUM_TIERS.